x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/remoteaccess/RemoteAccessAuthenticationServiceIntegTests.java

```diff
@@ -0,0 +1,229 @@
+                    )
+                )
+            ).writeToContext(threadContext);
+            authenticateAndAssertExpectedFailure(service, ex -> {
+                assertThat(ex, instanceOf(ElasticsearchSecurityException.class));
+                assertThat(
+                    ex.getCause().getMessage(),
+                    containsString(
+                        "role descriptor for remote access can only contain index privileges but other privileges found for subject"
+                    )
+                );
+            });
+        }
+
+        try (var ignored = threadContext.stashContext()) {
+            threadContext.putHeader(REMOTE_ACCESS_CLUSTER_CREDENTIAL_HEADER_KEY, "ApiKey " + encodedRemoteAccessApiKey);
+            Authentication authentication = AuthenticationTestHelper.builder().apiKey().build();
+            new RemoteAccessAuthentication(authentication, RoleDescriptorsIntersection.EMPTY).writeToContext(threadContext);
+            authenticateAndAssertExpectedFailure(service, ex -> {
+                assertThat(ex, instanceOf(ElasticsearchSecurityException.class));
+                assertThat(
+                    ex.getCause().getMessage(),
+                    containsString(
+                        "subject ["
+                            + authentication.getEffectiveSubject().getUser().principal()
+                            + "] has type ["
+                            + authentication.getEffectiveSubject().getType()
+                            + "] which is not supported for remote access"
+                    )
+                );
+            });
+        }
+    }
+
+    public void testSystemUserIsMappedToCrossClusterInternalRole() throws InterruptedException, IOException {
+        final String encodedRemoteAccessApiKey = getEncodedRemoteAccessApiKey();
+        final String nodeName = internalCluster().getRandomNodeName();
+        final ThreadContext threadContext = internalCluster().getInstance(SecurityContext.class, nodeName).getThreadContext();
+        final RemoteAccessAuthenticationService service = internalCluster().getInstance(RemoteAccessAuthenticationService.class, nodeName);
+
+        try (var ignored = threadContext.stashContext()) {
+            threadContext.putHeader(REMOTE_ACCESS_CLUSTER_CREDENTIAL_HEADER_KEY, "ApiKey " + encodedRemoteAccessApiKey);
+            new RemoteAccessAuthentication(
+                AuthenticationTestHelper.builder().internal(SystemUser.INSTANCE).build(),
+                new RoleDescriptorsIntersection(new RoleDescriptor("role", null, null, null, null, null, null, null))
+            ).writeToContext(threadContext);
+
+            final AtomicReference<Authentication> actual = new AtomicReference<>();
+            final CountDownLatch latch = new CountDownLatch(1);
+            service.authenticate(ClusterStateAction.NAME, new SearchRequest(), false, new LatchedActionListener<>(new ActionListener<>() {
```

Comment:
Similarly here, is `LatchedActionListener` necessary? The code is much more verbose with it.